### ISSUE TITLE
Update `@yamada-ui/switch` accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#1725]
### Description

This issue aims to address accessibility deficiencies in our @yamada-ui/switch component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

WAI-ARIA Roles, States, and Properties

- [ ] The switch has role [switch](https://w3c.github.io/aria/#switch).
- [ ]  The switch has an accessible label provided by one of the following:
  - [ ] A visible label referenced by the value of [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set on the element with role switch.
  - [ ] [aria-label](https://w3c.github.io/aria/#aria-label) set on the element with role switch.
- [ ] When on, the switch element has state [aria-checked](https://w3c.github.io/aria/#aria-checked) set to true.
- [ ] When off, the switch element has state [aria-checked](https://w3c.github.io/aria/#aria-checked) set to false.
- [ ] If a set of switches is presented as a logical group with a visible label, either:
  - [ ] The switches are included in an element with role [group](https://w3c.github.io/aria/#group) that has the property [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to the ID of the element containing the group label.
  - [ ] The set is contained in an HTML fieldset and the label for the set is contained in an HTML legend element.
- [ ] If the presentation includes additional descriptive static text relevant to a switch or switch group, the switch or switch group has the property [aria-describedby](https://w3c.github.io/aria/#aria-describedby) set to the ID of the element containing the description.

### Problem Statement/Justification

Improve accessibility

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by koralle** [comment]
I try it.
